h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -310,17 +359,28 @@ private static boolean hasMedian(AGG[] aggs) {
         return true;
     return false;
   }
-
-  private MRTask prepareMRFillTask(final G[] grps, final AGG[] aggs, final int medianCount) {
+  private MRTask prepareMRFillTask(final G[] grps, final AGG[] aggs, final GroupByColType[] gbColsTypes, final int medianCount) {
     return new MRTask() {
       @Override
       public void map(Chunk[] c, NewChunk[] ncs) {
         int start = (int) c[0].start();
         for (int i = 0; i < c[0]._len; ++i) {
           G g = grps[i + start];  // One Group per row
+          int gbColsCnt = g._gs.length + g._gsStr.length;
```

Comment:
can be replaced by gbColsTypes.length or eve better add g.colCount()

Reply:
Looks good. I changed it.